### DISCUSSION TITLE
Welcome to WheelMUD Discussions!

### DISCUSSION THREAD

**Discussion by DavidRieman** [DavidRieman/WheelMUD#20] [General]
## 👋 Welcome!
We’re using Discussions as a place to connect with other members of our community. We hope that you:
* Ask questions you’re wondering about.
* Share ideas.
* Engage with other community members.
* Welcome others and are open-minded. Remember that this is a community we build together 💪.

To get started, feel free to introduce yourself in a comment.  (For reference, there is a blurb about our current leadership [here](https://github.com/DavidRieman/WheelMUD/blob/master/Documentation/ProjectHistory.md#leadership).)

We are now also trying out a [WheelMUD Discord server](https://discord.gg/7smhbsNKRZ) to also facilitate banter with less overhead!

**Comment by JeremyTHolland** [comment]
Hi,
My name is Jeremy and have been tinkering with code from my early grade school days in the 80s with C64 and Apple II computers. More recently, I get paid to build .NET websites using Sitecore CMS. I was introduced to MUDs in college and really liked them, having grown up on interactive fiction from Infocom (Zork, etc.) the MUD world seemed like a logical extension of that genre. I'd like to build this project out into a flexible system to enable any sort of game. I'm partial to the Shadowrun setting as far as RPGs go I can bring that to the table as well as my coding skills. Looking forward to working with all of you!

**Comment by DavidRieman** [comment]
Welcome aboard, Jeremy!

My first coding was also on Apple II computers at school.

Building a flexible system is indeed what our focus is on.

We do plan to have at least one in-the-box reference system.  Right now it's Warrior Rogue Mage, but very under-developed on that front as the rules engine was pretty broken, based on unmaintained tech that we couldn't update with the rest of the framework.  We probably couldn't do anything like Shadowrun that would require licensing rights - but yeah all our older big discussions about how to build things in an extensible and flexible manner would assume a huge variety of settings, mechanics, etc.  It's an interesting challenge to avoid fundamental limitations like, say, "it must be Room-based"!

Anyway, let me know how you'd like to contribute here - Would you like to tinker around and fix random things, build some new feature, etc...  I'm certainly behind on documentation and maintainer stuff here but would be happy to focus on the bits first that will ensure you are unblocked to start having some fun with this  :)

**Reply by JeremyTHolland** [reply]
Maybe fixing a few things would give me a better overview than trying to implement a new feature. Do you have a few bugs or something specific in mind I could get my hands on? In the meantime I'm going to read all the docs to get my head around the current state of things. I'm open to suggestions though if you have a better idea to start out.

**Reply by DavidRieman** [reply]
Here are some issue ideas that shouldn't involve big framework changes to address:
https://github.com/DavidRieman/WheelMUD/issues/43 fix "password" change command
https://github.com/DavidRieman/WheelMUD/issues/44 combine "examine" / "look" commands
https://github.com/DavidRieman/WheelMUD/issues/46 add an admin console "connect" command
https://github.com/DavidRieman/WheelMUD/issues/67 render standard exits better

**Reply by ghost** [reply]
Hi everyone. I'm a child of the 80s so my evolutionary chart looks like this : 

Atari 400
Atari 1200XL
Atari 500ST
Amiga 500
Northgate 386SX   <-- this marks the day my inner-child died :P
Various beige boxes
Various beige boxes running Windows and Linux
Various beige boxes running Windows and Linux and OSX
Whatever it is my cloud stuff has to run on behind too many layers of abstraction to count

Like most of us from that era, I did my best to slog through Infocom games, wrote a bunch of BASIC, logged onto BBS systems starting at 300 baud and then worked my way up through various services including CompuServe, Genie, Prodigy, AOL...and here we are today. I did a little MUDing and rogue-like stuff but I think I missed the peak of that phenomenon during my college years. I'm still smitten by the written word. (How does the old quote go? "The greatest rendering engine is your imagination..." :)

I'm most  interested in the intersection of Infocom-style interactive fiction and MUDs and more about collaborative puzzle solving and story telling so that's probably where I'll start investigating WheelMUD.

**Reply by DavidRieman** [reply]
Welcome, @octave-syndicate!

Speaking of rogue-likes and MUDs... I've even seen rogue-like elements in MUD before (ages ago) and felt this is an under-represented combination.  I haven't looked for that lately, but it is one interesting area I hope our framework will be able to enable some day.

Years ago I thought about using WheelMUD as an integral part to a Puzzle Hunt that a friend was putting together for his org, but I couldn't commit to the short timeframe to put it together.  I still like the general idea and hope to pull it off the back-burner some day.  :)

Thanks for your first merged PR by the way!
PS: Remember to add yourself to `Credits.txt` if you like, in your next PR, no matter how small.

**Reply by ghost** [reply]
I'll add my _nom de plume_ to the credits the next time I make a PR!

I think the closest thing to a rogue-like MUD I recall playing is mangband but again my experience is pretty limited to occasional forays into the MU* world. One of the coolest MUDs I've played recently is https://writtenrealms.com ... the interface is open sources at https://github.com/teebes/herald and I think works against websockets under the hood. Could be an interesting bit of code to try to integrate with WheelMUD sometime around the 2.0 release. :P

For now I'll probably try to work a few PRs that reflect some minor challenges I've encountered along the way in my WheelMUD journey so far (StandardExits, unable to select talents with a space in the name, etc...small QOL stuff that might help to make a good impression when people come by the lot to kick the tires...) and maybe some documentation edits (It looks like mud.config might not still be a thing and that TestHarness is now ServerHarness...things like that.)

**Reply by DavidRieman** [reply]
Funny enough, I just noticed and fixed the TestHarness old name in docs and removed the mud.config entry last night, and https://github.com/DavidRieman/WheelMUD/pull/73 might finish addressing selecting things with spaces in names - would recommend checking that out.  For standard exits, you might want to look at https://github.com/DavidRieman/WheelMUD/issues/67 and self-assign it or comment there if you have something else related in mind  :)

**Comment by ghost** [comment]
I don't seem to have the ability to assign myself to https://github.com/DavidRieman/WheelMUD/issues/67

**Comment by DavidRieman** [comment]
Ah, right, I guess that's an org thing only.  Let's just comment on any issues we're looking into.

**Comment by Irthwynd** [comment]
Hello, David! I'm so glad I found your project, and I'm doubly glad you are actively working on it. I've been tinkering with C# and Java off and on with the end goal of creating a custom codebase MUD. I hope you don't mind if I watch your progress and try to unravel your coding logic. The process of and the components needed to create a MUD are still very mystifying.

**Reply by DavidRieman** [reply]
Welcome, Irthwynd!  Sure, of course I am happy this project struck your interest.  Feel free to lurk, check out the code, ask questions, etc - there are certainly ways to contribute and participate in this community from those of all skill levels - sometimes in the past we've had people just cruising the forums and contributing with feature design discussions and the like.

If you'd like to follow along with the code progressing, you might want to start by ensuring you are "following" and have "starred" the repository on github with the controls near the top.  Do note that our pace is not particularly _fast_ right now.  There are times WheelMUD has had no commits for nearly a year, and fairly recently came out of such a slump.  (And conversely in past years, there have been sometimes big flurries of activity from many developers.)

**Reply by Irthwynd** [reply]
Awesome! Understood about the progress. I have a good deal of experience on the world building end of things. It wasn't uncommon for large developmental lapses on both dev teams. On the engine side of things, I am 95% clueless where to even begin, so thank you for writing documentation. I'd love to contribute if and when I get to that point. I have a fully fleshed out world waiting to come to life, and I know more or less how I want to present it and how I want it to be played. Maybe I can translate some of those ideas into suggestions, although I ultimately understand this is your project.

**Reply by DavidRieman** [reply]
Cool.  For ideas/suggestions, you are welcome to post Ideas here, and I can work to see where applicable, reusable framework bits can get tracked in the Issues tracker where they feel they fit in the [rough roadmap](https://github.com/DavidRieman/WheelMUD/blob/master/docs/RoadMap.md).  Ideas can be fun to discuss, even when they don't fit in the confines of what should be "core", genre-agnostic functionality.

I've recently ported (and just updated) some of our "User Stories" design documents here to GitHub (from the offline forums of old).  One of them is the [On-Line Creation user story](http://htmlpreview.github.io/?https://github.com/DavidRieman/WheelMUD/blob/master/docs/Design/UserStory_OnLineCreation.htm).  If you have time, as an experienced builder:  It would be valuable if you could have a read through that and open a new discussion about it, if you have any feedback.  :)  For example, though designed to be flexible against the idea of developers constantly adding/updating Behaviors to the game, and allowing builders to do some advanced stuff without hitting the code, I do wonder if it's _too_ technical.

**Comment by fiedya** [comment]
Hi, hello :)
I got interested in mud written in c# and eventually found this. I'm going, just for fun, create small mud. Since I like c#, I found WheelMUD great startpoint...
Probably I will resign very soon, as always with projects, haha. But I would be glad if someone would help me with basics of this, because documentations and "tutorial" in it, has a few lacks :( I think.

**Reply by DavidRieman** [reply]
Hi, so sorry I missed my notifications on this somehow.  Let me know if you find your way back  :)  I tend to mostly only be active here when others are showing interest as well.

**Comment by chiefmultihat** [comment]
Hello,
I wanted to check up on an old project that's somewhere around 20 years or so old, I think :)  I'm IRay (David Ray) from the retired section.   Good to see WheelMUD still being maintained.

**Reply by DavidRieman** [reply]
Welcome back!  This weekend I just did some major fixes to document DB storage so if you pulled it down to check up on it, know that the basic world functionality should be in a much better state than it was a few days ago.

**Comment by jvoeller0129** [comment]
Its Good to see the code still out there, i was originally on the Early Dev Team and help Implement the Telnet Protocol and what not of days old. I used to go by cybershadow (cybrshdw) on wheelmud.net website

**Comment by jlechem** [comment]
HI everyone, I've been doing .net for 20 years, I contributed to this project before it was on GitHub. I recently forked the repo and am looking forward to contributing again!
